File: 0389-find-the-difference/0389-find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        shash = {}
        thash = {}

        for chs in s:
            if chs in shash:
                shash[chs] += 1
            else:
                shash[chs] = 1
                
        for cht in t:
            if cht in thash:
                thash[cht] += 1
            else:
                thash[cht] = 1
        for cht in thash.keys():
            if cht not in shash or thash[cht] != shash[cht]:
                return cht

File: 0389-find-the-difference/0389-find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        HashMap<Character, Integer> shash = new HashMap<>();
        HashMap<Character, Integer> thash = new HashMap<>();
        char[] chs = s.toCharArray();
        char[] cht = t.toCharArray();
        
        for(char ch : chs) {
            shash.merge(ch, 1, Integer::sum);
        }
        for(char ch : cht) {
            thash.merge(ch, 1, Integer::sum);
        }
        
        for(Map.Entry<Character, Integer> entry : thash.entrySet()) {
            if(!shash.containsKey(entry.getKey()) || shash.get(entry.getKey()) != entry.getValue()) {
                return entry.getKey();
            }
        }
        
        return 'a';
    }
}